llvm/lib/Analysis/MLInlineAdvisor.cpp

```diff
@@ -484,8 +484,7 @@ void MLInlineAdvisor::print(raw_ostream &OS) const {
   OS << "\n";
   OS << "[MLInlineAdvisor] FuncLevels:\n";
   for (auto I : FunctionLevels)
-    OS << (I.first->isDead() ? "<deleted>" : I.first->getFunction().getName())
-       << " : " << I.second << "\n";
+    OS << I.first->getFunction().getName() << " : " << I.second << "\n";
```

Comment:
nit: since this is for diags, it would be useful to keep track of functions that will be deleted. maybe we can keep a set of such functions? can't see that costing anything significant. Since the functions aren't killed, we can have a set of pointers, really. wdyt?